src/url.js

```diff
@@ -43,6 +43,33 @@ export function parseUrl(url) {
   return info;
 }
 
+/**
+ * Appends a query string field and value to a url. `key` and `value`
+ * will be ran through `encodeURIComponent` before appending.
+ * @param {string} url
+ * @param {string} key
+ * @param {string} value
+ * @return {string}
+ */
+export function addParamToUrl(url, key, value) {
+  const urlObj = parseUrl(url);
```

Comment:
Yeah, pretty much, except there's no need to fallback. Just need to look for `?`, `&` and `#`.